File: Java/Application/app/src/main/java/com/ixidev/simplenotepad/db/NotesDB.java
package com.ixidev.simplenotepad.db;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import android.content.Context;

import com.ixidev.simplenotepad.model.Note;

/**
 * Created by ixi.Dv on 20/06/2018.
 */
@Database(entities = Note.class, version = 1)
public abstract class NotesDB extends RoomDatabase {
    public abstract NotesDao notesDao();

    public static final String DATABSE_NAME = "notesDb";
    private static NotesDB instance;

    public static NotesDB getInstance(Context context) {
        if (instance == null)
            instance = Room.databaseBuilder(context, NotesDB.class, DATABSE_NAME)
                    .allowMainThreadQueries()
                    .build();
        return instance;
    }
}

File: Kotlin/Application/app/src/main/java/com/ixidev/simplenotepad/db/NotesDB.kt
package com.ixidev.simplenotepad.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.ixidev.simplenotepad.entity.Note

/**
 * Created by ixi.Dv on 20/06/2018.
 */
@Database(entities = [Note::class], version = 1)
abstract class NotesDB : RoomDatabase() {
    abstract fun notesDao(): NotesDao
    companion object {
        private const val DATABSE_NAME = "notesDb"
        private var instance: NotesDB? = null
        fun getInstance(context: Context): NotesDB {
            if (instance == null) instance =
                Room.databaseBuilder(
                    context,
                    NotesDB::class.java,
                    DATABSE_NAME
                )
                    .allowMainThreadQueries()
                    .build()
            return instance as NotesDB
        }
    }
}